src/detect-engine-hsbd.c

```diff
@@ -3631,6 +3839,345 @@ static int DetectEngineHttpServerBodyFileDataTest03(void)
     return result;
 }
 
+/*
+ * Tests buffer inspection like in ips mode,
+ * here all chunks size are smaller than the window_size.
+
+ * in:               buffer:
+ * [ab]              [ab]
+ * [cd]              [abcd]
+ * [ef]              [cdef]
+*/
+static int DetectEngineHttpServerBodyFileDataTest04(void)
+{
+
+    const char yaml[] = "\
+%YAML 1.1\n\
+---\n\
+libhtp:\n\
+\n\
+  default-config:\n\
+\n\
+    http-body-inline: yes\n\
+    response-body-minimal-inspect-size: 6\n\
+    response-body-inspect-window: 3\n\
+";
+
+    struct TestSteps steps[] = {
+        {   (const uint8_t *)"GET /index.html HTTP/1.0\r\n"
+            "Host: www.openinfosecfoundation.org\r\n"
+            "User-Agent: Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7\r\n"
+            "\r\n"
+            "HTTP/1.0 200 ok\r\n"
+            "Content-Type: text/html\r\n"
+            "Content-Length: 6\r\n"
+            "\r\n"
+            "ab",
+            0, STREAM_TOSERVER, 0 },
+        {   (const uint8_t *)"cd",
+            0, STREAM_TOSERVER, 0 },
+        {   (const uint8_t *)"ef",
+            0, STREAM_TOSERVER, 0 },
+        {   NULL, 0, 0, 0},
+    };
+
+    const char *sigs[] = {
+        "alert http any any -> any any (content:\"ab\"; file_data; sid:1;)",
```

Comment:
These rules are wrong: it should be file_data preceding content, not content followed by file_data.